test/JDBC/input/storedProcedures/Test-sp_set_session_context-dep-vu-verify.sql

```diff
@@ -0,0 +1,8 @@
+EXEC test_sp_set_session_context_proc;
```

Comment:
do we actually need a separate JDBC test? 
I think mix could cover the scenario, and it could help simplify your PR a little bit

Reply:
We might need both dependency and also version upgrade tests.
The dependency tests require that sp_set_session_context and session_context are defined in the prepare step, while in the version upgrade case they aren't defined.